fe/src/pages/ProductDetails/ProductDetails.tsx

```diff
@@ -0,0 +1,62 @@
+import { CheckCircleIcon, NotAllowedIcon } from '@chakra-ui/icons';
+import {
+  Flex, Heading, Image, Spacer, Text,
+} from '@chakra-ui/react';
+import React from 'react';
+// import { useParams } from 'react-router-dom';
+
+// interface RouteParams {
+//   id: string;
+// }
+
+const ProductDetails: React.FC = () => {
+  // @todo - Remove mock data and make a graphql hook to get single item based on product id
+  const data = {
+    product: {
+      id: 7,
+      name: 'Nikon 600P',
+      description:
+        'This is this product description..\n\nStats for HP Pavilion 300H:\nCamera - Good\nQuality - Very bad\n',
+      isAvailable: true,
+      imageUrl:
+        'https://img.theweek.in/content/dam/week/news/sci-tech/2019/June/camera-photographer-photo-technology-shut.jpg',
+    },
+    quantity: 4,
+  };
+  // const { id } = useParams<RouteParams>();
+
+  const { product, quantity } = data;
+  const {
+    description, isAvailable, name, imageUrl,
+  } = product;
+  return (
+    <Flex h="100%" justifyContent="center" alignItems="center">
+      <Flex
+        boxSizing="border-box"
+        marginX="2em"
+        flexDirection={{ base: 'column', lg: 'row' }}
+        textAlign="center"
+        justifyContent="center"
+      >
+        <Flex textAlign="center" justifyContent="center" mx="1em" mb="1em">
+          <Image w="90%" src={imageUrl} alt="Product Image Main" />
+        </Flex>
+        <Spacer />
+        <Flex flexDirection="column">
+          <Heading textAlign="center">{name}</Heading>
+          <Text whiteSpace="pre-wrap">{description}</Text>
+          <Flex flexDirection="row" justifyContent="flex-end" alignItems="center">
+            <Text mr="0.5em">{quantity}</Text>
+            {isAvailable ? (
+              <CheckCircleIcon w={8} h={8} color="green.500" />
+            ) : (
+              <NotAllowedIcon w={8} h={8} color="red.500" />
+            )}
+          </Flex>
+        </Flex>
+      </Flex>
+    </Flex>
```

Comment:
Have you heard of `Box` component? :P 
I feel like there is a little too much nesting here, for example the first 2 flex containers can be joined. Update it accordingly to the following points:

2 points about the styling in this page:

**1)** Please take into account that the content of description, and other properties can be longer than seen in fake data. Currently inputting some long content really messes up the look of the picture:

![picture size weird](https://user-images.githubusercontent.com/48998862/117439333-635e7c80-af3b-11eb-9e72-61e350ef6644.JPG)


**2)** Also could you update the styling a bit? See on mobile view there is a lot of unused space (red lines) and the picture seems small compared to scale of other things:

![space](https://user-images.githubusercontent.com/48998862/117438689-8dfc0580-af3a-11eb-96e1-fd454369870e.JPG)

Reply:
also make sure it can take random sized images and work correctly.

Reply:
I implemented functionality, so it will crop the image if it's to big. I used ChakraUI aspect ratio component [here](https://chakra-ui.com/docs/layout/aspect-ratio)